airflow/www/static/js/dag/details/taskInstance/Details.tsx

```diff
@@ -335,11 +351,49 @@ const Details = ({ gridInstance, taskInstance, group }: Props) => {
                       // skip
                     }
                   }
+                  const fieldStr = field as string;
+                  try {
+                    const fieldJson = JSON.parse(fieldStr);
+                    return (
```

Comment:
Oh let's not return inside of the `try/catch` block. Instead just assign the fieldJson variable inside try/catch and then add a check if its not null inside the jsx return

Reply:
This is a lot of logic now. How about we make a separate `RenderedFields` component and handle all of this logic there?

Reply:
Thanks for the hints, needed ~10 iterations but now I assume I got it. Consolidated and in total only 19 lines added in code-base for a common component - XCom following :-D